stan/math/prim/prob/double_exponential_cdf.hpp

```diff
@@ -32,66 +33,89 @@ template <typename T_y, typename T_loc, typename T_scale>
 
   T_partials_return cdf(1.0);
-  operands_and_partials<T_y, T_loc, T_scale> ops_partials(y, mu, sigma);
+  operands_and_partials<T_y_ref, T_mu_ref, T_sigma_ref> ops_partials(
+      y_ref, mu_ref, sigma_ref);
 
-  scalar_seq_view<T_y> y_vec(y);
-  scalar_seq_view<T_loc> mu_vec(mu);
-  scalar_seq_view<T_scale> sigma_vec(sigma);
-  size_t size_sigma = stan::math::size(sigma);
-  size_t N = max_size(y, mu, sigma);
+  const auto& y_col = as_column_vector_or_scalar(y_ref);
+  const auto& mu_col = as_column_vector_or_scalar(mu_ref);
+  const auto& sigma_col = as_column_vector_or_scalar(sigma_ref);
 
-  VectorBuilder<true, T_partials_return, T_scale> inv_sigma(size_sigma);
-  for (size_t i = 0; i < size_sigma; i++) {
-    inv_sigma[i] = inv(value_of(sigma_vec[i]));
-  }
+  const auto& y_arr = as_array_or_scalar(y_col);
+  const auto& mu_arr = as_array_or_scalar(mu_col);
+  const auto& sigma_arr = as_array_or_scalar(sigma_col);
 
-  VectorBuilder<true, T_partials_return, T_y, T_loc, T_scale> scaled_diff(N);
-  VectorBuilder<true, T_partials_return, T_y, T_loc, T_scale> exp_scaled_diff(
-      N);
+  ref_type_t<decltype(value_of(y_arr))> y_val = value_of(y_arr);
+  ref_type_t<decltype(value_of(mu_arr))> mu_val = value_of(mu_arr);
+  ref_type_t<decltype(value_of(sigma_arr))> sigma_val = value_of(sigma_arr);
 
-  for (size_t n = 0; n < N; n++) {
-    const T_partials_return y_dbl = value_of(y_vec[n]);
-    const T_partials_return mu_dbl = value_of(mu_vec[n]);
-    scaled_diff[n] = (y_dbl - mu_dbl) * inv_sigma[n];
-    exp_scaled_diff[n] = exp(scaled_diff[n]);
+  check_not_nan(function, "Random variable", y_val);
+  check_finite(function, "Location parameter", mu_val);
+  check_positive_finite(function, "Scale parameter", sigma_val);
 
-    if (y_dbl < mu_dbl) {
-      cdf *= exp_scaled_diff[n] * 0.5;
-    } else {
-      cdf *= 1.0 - 0.5 / exp_scaled_diff[n];
-    }
+  if (size_zero(y, mu, sigma)) {
+    return 1.0;
   }
 
-  for (size_t n = 0; n < N; n++) {
-    const T_partials_return y_dbl = value_of(y_vec[n]);
-    const T_partials_return mu_dbl = value_of(mu_vec[n]);
-    const T_partials_return sigma_dbl = value_of(sigma_vec[n]);
+  const auto& inv_sigma = to_ref(inv(sigma_val));
+  const auto& scaled_diff = to_ref_if<!is_constant_all<T_scale>::value>(
+      (y_val - mu_val) * inv_sigma);
+  const auto& exp_scaled_diff = to_ref(exp(scaled_diff));
 
-    const T_partials_return rep_deriv
-        = y_dbl < mu_dbl ? cdf * inv_sigma[n]
-                         : cdf * inv_sigma[n] / (2 * exp_scaled_diff[n] - 1);
-
-    if (!is_constant_all<T_y>::value) {
-      ops_partials.edge1_.partials_[n] += rep_deriv;
-    }
-    if (!is_constant_all<T_loc>::value) {
-      ops_partials.edge2_.partials_[n] -= rep_deriv;
+  size_t N = max_size(y, mu, sigma);
+  T_rep_deriv rep_deriv;
+  if (is_vector<T_y>::value || is_vector<T_loc>::value) {
+    using array_bool = Eigen::Array<bool, Eigen::Dynamic, 1>;
+    cdf = forward_as<array_bool>(y_val < mu_val)
```

Comment:
@andrjohns The comparison is here. It fails when we vave double and fvar<var> scalars in function arguments making these two compared values have scalars double and var. You can see whole error message on Jenkins.

Reply:
Thanks I'll have a look and see what I can see

Reply:
Thanks for the help. I managed to fix it.